starfish/image/_segmentation/watershed.py

```diff
@@ -12,15 +12,37 @@
 from starfish.imagestack.imagestack import ImageStack
 from starfish.munge import relabel
 from starfish.stats import label_to_regions
-from starfish.types import Indices
+from starfish.types import Indices, Number
 from ._base import SegmentationAlgorithmBase
 
 
 class Watershed(SegmentationAlgorithmBase):
-    """
-    Implements watershed segmentation.  TODO: (dganguli) FILL IN DETAILS HERE PLS.
-    """
-    def __init__(self, dapi_threshold, input_threshold, min_distance, **kwargs) -> None:
+
+    def __init__(
+        self,
+        dapi_threshold: Number,
+        input_threshold: Number,
+        min_distance: int,
+        **kwargs
+    ) -> None:
+        """Implements watershed segmentation of cells seeded from a DAPI image the the point clouds
```

Comment:
Implements watershed segmentation of cells. Algorithm is seeded by DAPI image. Binary segmentation mask is computed from a maximum projection of spots across C and R, which is subsequently thresholded.